### DISCUSSION TITLE
Is there any option to send MQTT msg to take camera snapshots

### DISCUSSION THREAD

**Discussion by rajalingamj175** [blakeblackshear/frigate#4319] [Ask A Question]
Need to take camera snapshots using MQTT msg

**Comment by NickM-27** [comment]
What do you mean by take snapshot? You can use the `<camera_name>/latest.jpg` and save that to take a snapshot

What is your usecase?

**Reply by rajalingamj175** [reply]
I want to get image on run time, once mqtt message is triggered from user event. 
usecase:
* when system is in ideal case user need to send one mqtt msg and get live image.

**Reply by NickM-27** [reply]
Yeah, sounds like you just want this https://docs.frigate.video/integrations/api#get-apicamera_namelatestjpgh300

**Reply by rajalingamj175** [reply]
Yes. But i need via MQTT. This is API call right?

**Reply by NickM-27** [reply]
How do you have access to MQTT and not HTTP? MQTT doesn't have responses to calls like HTTP so it would not work like it seems you're expecting (unless I am misunderstanding)

**Reply by rajalingamj175** [reply]
Yes correct. 

I cant access APIs, its in running local network.

In my setup i'm getting image when motion/event triggered in mqtt topic<frigate/*camera_name*/person/snapshot>. 
Like that if i send MQTT msg manually can i get live frame image using different or same mqtt topic.

Scenario:
In my setup i'm using cloud MQTT, i just listen the events in cloud. If i get any event like *person count* <frigate/*cameraname*/all>, it should automatically call the MQTT and get the current image.

**Reply by NickM-27** [reply]
> Scenario: In my setup i'm using cloud MQTT, i just listen the events in cloud. If i get any event like _person count_ <frigate/_cameraname_/all>, it should automatically call the MQTT and get the current image.

That's not how MQTT works though, you can't make a call and get a response. In any case, you can use https://docs.frigate.video/integrations/mqtt/#frigatecamera_nameobject_namesnapshot

**Reply by rajalingamj175** [reply]
Thanks Nick. This is the way i'm getting event snapshot
